cmd/appengine/device.go

```diff
@@ -1634,34 +1636,41 @@ func parseSendDataPayload(payload string, mappingType interfaces.AstarteMappingT
 	case interfaces.BinaryBlobArray, interfaces.BooleanArray, interfaces.DateTimeArray, interfaces.DoubleArray,
 		interfaces.IntegerArray, interfaces.LongIntegerArray, interfaces.StringArray:
 
-		//wait it's all string?
-		payload = strings.Replace(payload, "[", "", -1)
-		payload = strings.Replace(payload, "]", "", -1)
-		payload_parsed := strings.Split(payload, ",")
-		//always has been
+		//check if payload is an empty array, bypass conversions and just return an empty array
+		// if it is not, proceed as usual
+		if payload == "[]" {
+			ret = make([]interface{}, 0)
+		} else {
+			retArray := []interface{}{}
+			payload = strings.TrimSpace(payload)
+			if pass, err := config.RegexCheck(payload, `^\[.*\]$`); !pass {
+				return nil, err
+			}
```

Comment:
It looks like the `config.RegexCheck` funcition is just a wrapper around `regexp.MatchString`.
What about changing to the following? In this way we don't add new helper functions.
```suggestion
			if pass, err := regexp.MatchString(`^\[.*\]$`, payload); !pass {
				return nil, err
			}
```